### PR TITLE
strip yaml extension from latest file path when publishing

### PR DESCRIPTION
**Pull request by nkubala**
@dlorenc @sharifelgamal

### CODE REVIEW

File: runtime_builders/publish_builders.py [hunk 1]
```diff
@@ -41,7 +41,16 @@ def main():
                     project_name = config['project']
                     latest = config['latest']
                     prefix = builder_util.RUNTIME_BUCKET_PREFIX
-                    latest_file = latest[len(prefix+project_name+'-'):]
+                    if not latest.startswith(prefix):
+                        logging.error('Please provide fully qualified '
+                                      'path to config file in GCS!')
```
Comment (znewman01) [inline runtime_builders/publish_builders.py]: Might be nice to also say:

    path should start with ....

(and below)

File: runtime_builders/publish_builders.py [hunk 2]
```diff
@@ -41,7 +41,16 @@ def main():
...
+                        sys.exit(1)
+                    if not latest.endswith('.yaml'):
+                        logging.error('Please provide yaml config file to '
+                                      'publish as latest!')
+                        sys.exit(1)
+                    latest_file = latest[len(prefix+project_name+'-') \
+                                         :-(len('.yaml'))]
```
Comment (dlorenc) [inline runtime_builders/publish_builders.py]: Could you use os.path.splitext for checking that it ends with .yaml and removing it?

File: runtime_builders/publish_builders.py [hunk 3]
```diff
@@ -41,7 +41,16 @@ def main():
...
-                    latest_file = latest[len(prefix+project_name+'-'):]
+                    if not latest.startswith(prefix):
+                        logging.error('Please provide fully qualified '
+                                      'path to config file in GCS!')
+                        sys.exit(1)
+                    if not latest.endswith('.yaml'):
+                        logging.error('Please provide yaml config file to '
+                                      'publish as latest!')
+                        sys.exit(1)
+                    latest_file = latest[len(prefix+project_name+'-') \
```
Comment (znewman01) [inline runtime_builders/publish_builders.py]: at this point, worth a regex for readability (or at least factor out "prefix+project_name+'-')

File: runtime_builders/publish_builders.py (hunk 2 above)
Comment (nkubala) [inline runtime_builders/publish_builders.py]: done

**Pull request merged**